eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -169,8 +169,8 @@ object Commitments {
       return Failure(ExpiryTooBig(commitments.channelId, maximum = maxExpiry, actual = cmd.cltvExpiry, blockCount = blockHeight))
     }
 
-    if (cmd.amount < commitments.remoteParams.htlcMinimum) {
-      return Failure(HtlcValueTooSmall(commitments.channelId, minimum = commitments.remoteParams.htlcMinimum, actual = cmd.amount))
+    if (cmd.amount < commitments.remoteParams.htlcMinimum.max(1 msat)) {
```

Comment:
Why do we need this if we have a hard requirement in the settings?

Reply:
Because we can't ensure that our remote has that requirement as well ;)
It seems that c-lightning's default value here is 0, but we still don't want to forward them 0-value HTLCs.